model_signing/hashing/file_test.py

```diff
@@ -27,21 +27,21 @@
 _UNUSED_PATH = pathlib.Path("unused")
 
 
-@pytest.fixture(scope="session")
+@pytest.fixture(scope="class")
 def sample_file(tmp_path_factory):
     file_path = tmp_path_factory.mktemp("dir") / "text.txt"
     file_path.write_text(_FULL_CONTENT)
     return file_path
 
 
-@pytest.fixture(scope="session")
+@pytest.fixture(scope="class")
 def sample_file_content_only(tmp_path_factory):
     file_path = tmp_path_factory.mktemp("dir") / "text.txt"
     file_path.write_text(_CONTENT)
     return file_path
 
 
-@pytest.fixture(scope="session")
+@pytest.fixture(scope="class")
```

Comment:
For these expected digests ones, does it matter if the expected digest lasts longer? It's not like we expect the value to change based on which test class calls it?

Reply:
They're tied with these files, it does not make sense to have them around once these files are no longer present.

These fixtures should only exist while testing the hashing layer, once we move to the serialization layer we have a different set of fixtures for the models. There, because we alter the models, we use the default scope, where each fixture is created before each test and destroyed at the end of each one. We don't mutate these here, so test scope is too much. Next possible one is class scope.

Reply:
Ah, I had wrongly assumed this fixture was used elsewhere